e2e/specs/st_multiselect.spec.js

```diff
@@ -212,37 +187,109 @@ describe("st.multiselect", () => {
                 "value 7: []" +
                 "value 8: []" +
                 "value 9: []" +
+                "value 10: []" +
+                "value 11: []" +
                 "multiselect changed: False"
             );
         });
       });
     });
 
     it("calls callback if one is registered", () => {
-      cy.get(".stMultiSelect")
-        .should("have.length", 9)
-        .last()
+      cy.getIndexed(".stMultiSelect", 10)
         .find("input")
         .click();
       cy.get("li")
         .first()
         .click();
 
       cy.get("[data-testid='stText']")
-        .should("have.length", 10)
         .should(
-          "have.text",
-          "value 1: []" +
-            "value 2: ['female']" +
-            "value 3: []" +
-            "value 4: ['tea', 'water']" +
-            "value 5: []" +
-            "value 6: []" +
-            "value 7: []" +
-            "value 8: []" +
-            "value 9: ['male']" +
+          "contain.text",
+            "value 11: ['male']" +
             "multiselect changed: True"
         );
     });
   });
+
+  describe("when using max_selections for st.multiselect", () => {
+    beforeEach(() => {
+      // this is the 'close button' element for 'Male'
+      cy.get('.stMultiSelect span[data-baseweb="tag"] span:last-child')
+        .eq(0)
+        .click();
+    });
+    it("should show the correct text when maxSelections is reached not within a form and closing input", () => {
+      selectNthOptionForKthMultiselect(0, 8)
+      cy.getIndexed(".stMultiSelect", 8).then(el => {
+        cy
+          .wrap(el)
+          // recheck that we have hit the limit
+          .find("input")
+          .click()
+          .get("li")
+          .should("have.text", "You can only select up to 1 option. Remove an option first.")
+      });
+    });
+
+    it("should show the correct text when maxSelections is reached not within a form and not closing input", () => {
+      cy.getIndexed(".stMultiSelect", 8)
+        .find("input")
+        .click();
+      cy
+        .get("li")
+        .eq(0)
+        .click();
```

Comment:
nit: should be able to use the newly added helper function for this

Reply:
The helper function also closes the multiselect. I'll add a new method that explicit says it's not closing and refactor this.